src/client/components/series-menu/series-menu.tsx

```diff
@@ -57,32 +58,24 @@ export class SeriesMenu extends React.Component<SeriesMenuProps, SeriesMenuState
 
   globalKeyDownListener = (e: KeyboardEvent) => enterKey(e) && this.onOkClick();
 
-  saveFormat = (format: SeriesFormat) => this.setState({ format });
+  saveSeries = (series: Series) => this.setState({ series });
 
   onCancelClick = () => this.props.onClose();
 
   onOkClick = () => {
     if (!this.validate()) return;
     const { series: originalSeries, clicker, essence, onClose } = this.props;
-    const series = this.constructSeries();
-    clicker.changeSeriesList(essence.series.replaceSeries(originalSeries, series));
+    clicker.changeSeriesList(essence.series.replaceSeries(originalSeries, this.state.series));
     onClose();
   }
 
   validate() {
-    const series = this.constructSeries();
-    return !this.props.series.equals(series);
-  }
-
-  private constructSeries() {
-    const { series } = this.props;
-    const { format } = this.state;
-    return series.set("format", format);
+    return !this.props.series.equals(this.state.series);
```

Comment:
Sometimes, inner menu could yield invalid series (for example missing operand for difference between measures) and we need to check this here ...